detox/src/logger/utils/DetoxLogFinalizer.test.js

```diff
@@ -172,21 +172,10 @@ describe('DetoxLogFinalizer', () => {
       expect(fs.existsSync(temporaryFiles[2])).toBe(false);
     });
 
-    it('should not create logs for successful run when configured', async () => {
+    it('should create logs even for successful run when configured to keep failing logs', async () => {
       session.detoxConfig.artifacts.plugins.log.keepOnlyFailedTestsArtifacts = true;
       session.testResults = [{ success: true }];
 
-      await createLogFiles();
-      finalizer.finalizeSync();
-      expect(fs.existsSync(artifactsDir())).toBe(false);
-      expect(fs.existsSync(temporaryFiles[1])).toBe(false);
-      expect(fs.existsSync(temporaryFiles[2])).toBe(false);
-    });
-
-    it('should create logs for failing run when configured', async () => {
```

Comment:
Don't you want to keep this test case? It's not a given IMO, even though it's all the same as `{success: true}` implementation-wise.

Reply:
Well, since the coverage is 100% there, adding another if-else branch will reduce it, forcing you to add a new test, so hopefully this is not very critical.